src/core/canvasselectedboxesactions.cpp

```diff
@@ -864,6 +864,14 @@ void Canvas::alignSelectedBoxes(const Qt::Alignment align,
         skip = mLastSelectedBox;
         geometry = mLastSelectedBox->getAbsBoundingRect();
         break;
+    case AlignRelativeTo::lastSelectedPivot:
+        if(!mLastSelectedBox) return;
+        skip = mLastSelectedBox;
+        geometry = QRectF(mLastSelectedBox->getPivotAbsPos(),mLastSelectedBox->getPivotAbsPos());
+        break;
+    case AlignRelativeTo::boundingBox:
+        geometry = QRectF(0., 0., mWidth, mHeight); // TODO: not using it?
```

Comment:
Todo what? :)

Reply:
I would need to double check if it is used or not for aligning object to canvas... just that, I'll try removing it and see if it breaks something...

Reply:
Ok, let's get the boudingbox stuff working 100% before the UI stuff.

Is everything working as expected? I did basic testing of some boxes and it seems to work, but I didn't test everything.

Reply:
Yes, you are right.

I test it pretty deeply and the only two issues I found are the ones I commented:
- pivot alignment crashes Friction with Null object, I need to check if you fixed it by removing a function, you already commented about it
- text object doesn't work perfectly, it seems to have a different approach for its origin... try and let me know if you understand why

Reply:
> pivot alignment crashes Friction with Null object, I need to check if you fixed it by removing a function, you already commented about it

This is fixed, you tried to access stroke settings without checking that settings was valid `if (settings)`. It wasn't used for anything anyway, so I removed it.

> text object doesn't work perfectly, it seems to have a different approach for its origin... try and let me know if you understand why

Ok, will need to test.

Reply:
I've been doing some tests and this seems not to make any difference so I guess the whole case can be deleted.

Do you want me to do it?